distributed/comm/tcp.py

```diff
@@ -299,16 +299,25 @@ async def write(self, msg, serializers=None, on_error="message"):
             # trick to enque all frames for writing beforehand
             for each_frame_nbytes, each_frame in zip(frames_nbytes, frames):
                 if each_frame_nbytes:
-                    if stream._write_buffer is None:
-                        raise StreamClosedError()
+                    each_frame = memoryview(each_frame)
```

Comment:
Would use `ensure_memoryview` as this handles normalizing `memoryview`s to be 1-D, contiguous and `uint8` typed (as other code like in Tornado often assumes).

Can also move the comment from below.

```suggestion
                    # Make sure that `len(data) == data.nbytes`
                    # See <https://github.com/tornadoweb/tornado/pull/2996>
                    each_frame = ensure_memoryview(each_frame)
```